File: samples/AdventureWorksLT/UdtFlagAttribute.cs
using DevZest.Data;
using DevZest.Data.Addons;
using DevZest.Data.Annotations.Primitives;

namespace DevZest.Samples.AdventureWorksLT
{
    [ModelMemberAttributeSpec(typeof(ColumnNotNull), true, typeof(_Boolean))]
    public sealed class UdtFlagAttribute : ColumnAttribute
    {
        protected override void Wireup(Column column)
        {
            if (column is _Boolean boolean)
                boolean.Nullable(false);
        }
    }
}

File: samples/AdventureWorksLT.VB/UdtFlagAttribute.vb
Imports DevZest.Data.Annotations.Primitives
Imports DevZest.Data.Addons

<ModelDesignerSpec(New Type() {GetType(ColumnNotNull)}, New Type() {GetType(_Boolean)})>
Public NotInheritable Class UdtFlagAttribute
    Inherits ColumnAttribute

    Protected Overrides Sub Wireup(column As Column)
        If TypeOf column Is _Boolean Then
            column.Nullable(False)
        End If
    End Sub
End Class
